src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Unix.cs

```diff
@@ -25,24 +37,28 @@ public DriveType DriveType
             [SecuritySafeCritical]
             get
             {
-                Interop.libc.statfs data;
-                Interop.ErrorInfo errorInfo;
-                if (Interop.libc.TryGetStatFsForDriveName(Name, out data, out errorInfo))
+                DriveType type;
+                int result = Interop.Sys.GetFormatInfoForMountPoint(Name, out type);
+                if (result == 0)
                 {
-                    return GetDriveType(Interop.libc.GetMountPointFsType(data));
+                    return type;
                 }
-
-                // This is one of the few properties that doesn't throw on failure,
-                // instead returning a value from the enum.
-                switch (errorInfo.Error)
+                else
                 {
-                    case Interop.Error.ELOOP:
-                    case Interop.Error.ENAMETOOLONG:
-                    case Interop.Error.ENOENT:
-                    case Interop.Error.ENOTDIR:
-                        return DriveType.NoRootDirectory;
-                    default:
-                        return DriveType.Unknown;
+                    Interop.ErrorInfo err = Interop.Sys.ConvertErrorPlatformToPal(result).Info();
```

Comment:
Here too: `var errorInfo = new Interop.ErrorInfo(result)`; 

Why did you shorten errorInfo to err?

Also, maybe I just need to read further, but curious why we used errno/GetLastError before but now return the error.